GitUI/UserControls/RevisionGrid/FilterInfo.cs

```diff
@@ -318,6 +281,8 @@ public void ResetAllFilters()
             ByPathFilter = false;
             ByBranchFilter = false;
             ShowCurrentBranchOnly = false;
+            ShowOnlyFirstParent = false;
+            ShowMergeCommits = true;
```

Comment:
❔
```suggestion
            ShowMergeCommits = true;
            
            // ShowFullHistory and ShowSimplifyMerges are not filters which affect HasFilter
```

Reply:
There is a note in GetSummary(). Maybe this is better added to HasFilter?
Moving ResetAllFilters() to just after HasRevisionFilter, they are related and often updated together.